enumerable.rb

```diff
@@ -1,50 +1,88 @@
 module Enumerable
-    def my_each
-        if block_given?
-            for x in self do
-                yield (x)
-            end
-        end
-        return self
-    end
-    
-    def my_each_with_index
-        if block_given?
-            for x in self do
-                yield (x)                
-            end
-            for y in self do
-                y.index(self[x]).to_i
-            end
-        end
-        return self
-    end
-    ['a', 'b', 'c'].my_each_with_index { |el, i| puts i }
-    def my_select
-
+  def my_each
+    if block_given?
+      each do |x|
+        yield x
+      end
     end
+    self
+  end
 
-    def my_all?
-
+  def my_each_with_index
+    if block_given?
+      idx = 0
+      each do |x|
+        yield(x, idx)
+        idx += 1
+      end
     end
+    self
+  end
 
-    def my_any?
+  def my_select
+    return unless block_given?
 
+    array = []
+    each do |x|
+      array.push(self[x]) if yield(self[x]) == true
     end
+    array
+  end
 
-    def my_none?
+  def my_all?
+    my_each { |x| return false if yield(x) == false }
+    true
+  end
 
-    end
+  def my_any?
+    my_each { |x| return true if yield(x) == true }
+    false
+  end
 
-    def my_count
+  def my_none?
+    my_each { |x| return false if yield(x) == true }
+    true
+  end
```

Comment:
#### my_none? 

- [ ] when no block or argument is given returns true only if none of the collection members is true

     > this [nil, false, true, []].my_none? should be equal to this [nil, false, true, []].none? => false

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class

     > this [1, 2.5, 'a', 9].my_none?(Integer) should be equal to this [1, 2.5, 'a', 9].none?(Integer) => fakse

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex

     > %w[dog door rod blade].my_none?(/z/) should be equal to this %w[dog door rod blade].none?(/z/) => false

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

     > [3,4,7,11].my_none?(3) should be equal to this [3,4,7,11].none?(3) => false

[OPTIONAL]

I think if we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?